subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/SetupLoggingActionExecuter.java

```diff
@@ -30,26 +33,40 @@
 public class SetupLoggingActionExecuter implements BuildExecuter {
     private final BuildExecuter delegate;
     private final LoggingManagerInternal loggingManager;
+    private final ParallelExecutionManager parallelExecutionManager;
 
-    public SetupLoggingActionExecuter(BuildExecuter delegate, LoggingManagerInternal loggingManager) {
+    public SetupLoggingActionExecuter(BuildExecuter delegate, LoggingManagerInternal loggingManager, ParallelExecutionManager parallelExecutionManager) {
         this.delegate = delegate;
         this.loggingManager = loggingManager;
+        this.parallelExecutionManager = parallelExecutionManager;
     }
 
     @Override
     public Object execute(BuildAction action, BuildRequestContext requestContext, BuildActionParameters actionParameters, ServiceRegistry contextServices) {
         StartParameter startParameter = action.getStartParameter();
         loggingManager.setLevelInternal(startParameter.getLogLevel());
-        if (startParameter.isParallelProjectExecutionEnabled()) {
-            loggingManager.setMaxWorkerCount(startParameter.getMaxWorkerCount());
-        } else {
-            loggingManager.setMaxWorkerCount(1);
-        }
+        setupLogging(startParameter);
+        ParallelismConfigurationListener listener = new ParallelismConfigurationListener() {
+            @Override
+            public void onConfigurationChange(ParallelismConfiguration parallelismConfiguration) {
+                setupLogging(parallelismConfiguration);
+            }
+        };
+        parallelExecutionManager.addListener(listener);
         loggingManager.start();
         try {
             return delegate.execute(action, requestContext, actionParameters, contextServices);
         } finally {
             loggingManager.stop();
+            parallelExecutionManager.removeListener(listener);
+        }
+    }
+
+    private void setupLogging(ParallelismConfiguration parallelismConfiguration) {
+        if (parallelismConfiguration.isParallelProjectExecutionEnabled()) {
```

Comment:
No, the two settings are orthogonal. Parallel project execution (`--parallel`) just tells Gradle to assume that it is safe to run tasks from different projects in parallel. Independently of that Gradle will always do stuff in parallel (download dependencies, generate test reports, compile native files and tasks using the `WorkerExecutor` API). That parallelism is configured via `--max-workers`. This will change in 4.1

This piece of code is currently here because right now there is not enough happening in parallel by default yet to warrant showing all workers by default. Most of them would be idle in a typical project.